server/src/main/java/org/elasticsearch/action/search/SearchTask.java

```diff
@@ -53,4 +57,36 @@ public final SearchProgressListener getProgressListener() {
         return progressListener;
     }
 
+    public Supplier<SearchResponseMerger> getSearchResponseMergerSupplier(
```

Comment:
this could also be package private instead of public?

Reply:
package private doesn't work because AsyncSearchTask needs to override it and it is in a different package (xpack). But `protected` seems to work, so I'll adjust it to that.